### PR TITLE
Open Files needs to be more robust in error/notification

### PR DESCRIPTION
**Pull request by QilongTang**
### Purpose
MAGN-9552
Open Files needs to be more robust in error/notification

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
I got some excel tests failing, but they seem to fail on master as well.
- [ ] Snapshot of UI changes, if any.

### Reviewers

@ramramps 

Unified the way open file at different location handle exceptions. There might be further code change depending on if me and Neal could find more cases.

### FYIs

@jnealb

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs [hunk 1]
```diff
@@ -1124,7 +1124,15 @@ private void Open(object parameters)
...
-                model.Logger.Log(Resources.MessageFailedToOpenFile + e.Message);
+                if (e is FileNotFoundException)
+                {
+                    System.Windows.MessageBox.Show(String.Format(Resources.MessageFileNotFound, xmlFilePath));
+                }
+                else if (e is System.Xml.XmlException)
+                {
+                    System.Windows.MessageBox.Show(String.Format(Resources.MessageFailedToOpenFile, xmlFilePath));
+                }
+                model.Logger.Log(String.Format(Resources.MessageFailedToOpenFile, xmlFilePath, "\n"));
```
Comment (ramramps) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs]: you should e.,message here

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs [hunk 2]
```diff
@@ -1106,10 +1106,10 @@ private void Open(object parameters)
         {
             // try catch for exceptions thrown while opening files, say from a future version, 
             // that can't be handled reliably
+            string xmlFilePath = string.Empty;
+            bool forceManualMode = false; 
             try
             {
-                string xmlFilePath = string.Empty;
-                bool forceManualMode = false;
```
Comment (ramramps) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs]: Any reason for moving forceManulMode outside try block

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs (hunk 1 above)
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs]: We out put e.message on the next line.

Comment (ramramps) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs]: I see. model.Logger.Log(e) => this outputs the message?

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs (hunk 2 above)
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs]: hmm.. For system default variable declaration, I see no point of putting it in the try block. What do you think?

Comment (ramramps) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs]: thats fine. declarations can be outside.

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs (hunk 1 above)
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs]: Yes.

**Pull request merged**